client/messaging/external-link-dialog.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { MouseEvent, useCallback } from 'react'
+import styled from 'styled-components'
+import { closeDialog } from '../dialogs/action-creators'
+import { CommonDialogProps } from '../dialogs/common-dialog-props'
+import { RaisedButton, TextButton } from '../material/button'
+import { Dialog } from '../material/dialog'
+import { useAppDispatch, useAppSelector } from '../redux-hooks'
+import { mergeLocalSettings } from '../settings/action-creators'
+import { body2 } from '../styles/typography'
+
+const LinkAsText = styled.span`
+  ${body2};
+  overflow-wrap: anywhere;
+  user-select: text;
```

Comment:
It's kind of weird to only have this part as selectable. I'd prefer if the whole text was selectable, or none of it.

I think making the link selectable is a good idea, in case the users want to open it in non-default browser / incognito mode, or whatever.

Reply:
>I think making the link selectable is a good idea

That is the point - only link is selectable, so no accidental selection of adjacent text. Though it can be improved with click-select-copy behavior. What must be improved - long link handling, current implementation (with simple text truncation) just defeats the purpose of the link being selectable.